bin/extract_all_gtn_tutorials.py

```diff
@@ -0,0 +1,333 @@
+    tuto["feedback_mean_note"] = None
+    title = tuto["title"]
+    if title in feedback:
+        tuto["feedback_number"] = feedback[title]["number"]
+        tuto["feedback_mean_note"] = feedback[title]["mean note"]
+
+
+def get_visit_results(url: str, tuto: dict, plausible_api: str) -> None:
+    """
+    Extract visit results from Plausible URL
+    """
+    headers = {
+       'Authorization' : f"Bearer {plausible_api}"
+    }
+    results = get_request_json(url, headers)
+    if "results" in results:
+        for metric in ["visitors", "pageviews", "visit_duration"]:
+            tuto[metric] += results["results"][metric]["value"]
+
+
+def get_visits(tuto: dict, plausible_api: str) -> None:
+    """
+    Extract tutorial visitors and pageviews from Plausible
+    """
+    tuto["visitors"] = 0
+    tuto["pageviews"] = 0
+    tuto["visit_duration"] = 0
+    url_prefix = f"https://plausible.galaxyproject.eu/api/v1/stats/aggregate?site_id=training.galaxyproject.org&period=custom&date={tuto['pub_date']},{date.today()}&metrics=visitors,pageviews,visit_duration&filters=event:page%3D%3D%2Ftraining-material%2Ftopics%2F{tuto['topic_name']}%2Ftutorials%2F{tuto['tutorial_name']}"
+    if tuto["hands_on"]:
+        url = f"{url_prefix}%2Ftutorial.html"
+        get_visit_results(url, tuto, plausible_api)
+    if tuto["slides"]:
+        url = f"{url_prefix}%slides.html"
+        get_visit_results(url, tuto, plausible_api)
+
+
+def get_youtube_stats(tuto: dict) -> None:
+    """
+    Get YouTube stats
+    """
+    tuto["video_versions"] = 0
+    tuto["video_view"] = 0
+    ydl_opts = {
+        "ignoreerrors": True,
+        "quiet": True
+    }
+    if "video_library" in tuto and tuto["video_library"]["tutorial"]:
+        tuto["video_versions"] = len(tuto["video_library"]["tutorial"]["versions"])
+        for v in tuto["video_library"]["tutorial"]["versions"]:
+            url = f"https://www.youtube.com/watch?v={v['link']}"
+            with yt_dlp.YoutubeDL(ydl_opts) as ydl:
+                info = ydl.extract_info(url, download=False)
+                info = ydl.sanitize_info(info)
+                if info:
+                    tuto["video_view"] += info["view_count"]
+
+ 
+def format_tutorial(tuto: dict, edam_ontology, tools: dict, feedback: dict, plausible_api: str) -> None:
+    tuto["url"] = f'https://training.galaxyproject.org/{tuto["url"]}'
+    tuto["mod_date"] = format_date(tuto["mod_date"])
+    tuto["pub_date"] = format_date(tuto["pub_date"])
+    add_supported_servers(tuto)
+    get_short_tool_ids(tuto)
+    get_edam_topics(tuto, edam_ontology)
+    get_edam_operations(tuto, tools)
+    get_visits(tuto, plausible_api)
+    get_feedback(tuto, feedback)
+    get_youtube_stats(tuto)
+    return tuto
+
+
+def get_feedback_per_tutorials() -> Dict:
+    """
+    Get feedback from GTN API and group per tutorial
+    """
+    feedback = get_request_json("https://training.galaxyproject.org/training-material/api/feedback.json")
```

Comment:
feedback2.yaml is now a 'feature', it might be easier to parse.

```yaml
---
introduction:
  galaxy-intro-101:
  - rating: 5
    pro: very detailed and easy to follow, even for a complete beginner - great!
    con:
    date: '2018-09-14'
    lang:
  - rating: 5
```

*but* it is not copied to the API, doing so could be a very small PR to fix this line: https://github.com/galaxyproject/training-material/blob/main/.github/workflows/deploy.yml#L76

Reply:
Or as a slightly more complicated but nicer PR, add it to the `_plugins/api.rb` around L140

```
      # Feedback Data
      page2 = PageWithoutAFile.new(site, '', 'api/', 'feedback2.json')
      page2.content = JSON.generate(site.data['feedback2'])
      page2.data['layout'] = nil
      site.pages << page2
```

so you get back JSON and not yaml.